File: bidragsevne/src/main/java/no/nav/bidrag/beregn/felles/bidragsevne/beregning/Bidragsevneberegning.java
package no.nav.bidrag.beregn.felles.bidragsevne.beregning;

import no.nav.bidrag.beregn.felles.bidragsevne.bo.BeregnBidragsevneGrunnlagPeriodisert;

public interface Bidragsevneberegning {
    ResultatBeregning beregn(
        BeregnBidragsevneGrunnlagPeriodisert beregnBidragsevneGrunnlagPeriodisert);

    Double beregnMinstefradrag(
        BeregnBidragsevneGrunnlagPeriodisert beregnBidragsevneGrunnlagPeriodisert);

    Double beregnSkattetrinnBelop(
        BeregnBidragsevneGrunnlagPeriodisert beregnBidragsevneGrunnlagPeriodisert);

    static Bidragsevneberegning getInstance(){
        return new BidragsevneberegningImpl();
    }


}

File: bidrag-beregn-særtilskudd/src/main/kotlin/no/nav/bidrag/beregn/bidragsevne/beregning/BidragsevneBeregning.kt
package no.nav.bidrag.beregn.bidragsevne.beregning

import no.nav.bidrag.beregn.bidragsevne.bo.GrunnlagBeregning
import no.nav.bidrag.beregn.bidragsevne.bo.ResultatBeregning
import java.math.BigDecimal

interface BidragsevneBeregning {

    fun beregn(grunnlag: GrunnlagBeregning): ResultatBeregning

    fun beregnMinstefradrag(
        grunnlag: GrunnlagBeregning,
        minstefradragInntektSjablonBelop: BigDecimal,
        minstefradragInntektSjablonProsent: BigDecimal,
    ): BigDecimal

    fun beregnSkattetrinnBelop(grunnlag: GrunnlagBeregning): BigDecimal

    companion object {
        fun getInstance(): BidragsevneBeregning = BidragsevneBeregningImpl()
    }
}
